### ISSUE TITLE
How to upload file by using API Keys Authorization(while not oAuth2.0)?

### ISSUE THREAD

**Issue by xiebruce** [googleapis/google-api-php-client#1740]
My Api Key:
![image](https://user-images.githubusercontent.com/13575500/68359508-053eb680-0157-11ea-81fd-f2e1ff95a3d7.png)

My code:
```
use Google_Service_Drive_DriveFile;
use Google_Client;
use Google_Service_Drive;

$client = new Google_Client();
$client->setApplicationName("PicUploader");
$client->setDeveloperKey('<YOUR API KEY>');
$service = new Google_Service_Drive($client);

$file = new Google_Service_Drive_DriveFile([
	'name' => 'test.jpg',
	'mime' => 'image/jpeg'
]);
$result = $service->files->create($file, [
	'data' => file_get_contents('/path/to/test.jpg'),
	'mimeType' => 'image/jpeg',
	'uploadType' => 'multipart',
	'fields' => 'id',
]);
var_dump($result);exit;
```

Error:
```
Fatal error: Uncaught Google_Service_Exception: {
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "required",
    "message": "Login Required",
    "locationType": "header",
    "location": "Authorization"
   }
  ],
  "code": 401,
  "message": "Login Required"
 }
}
 in /Users/bruce/www/personal/PicUploader/vendor/google/apiclient/src/Google/Http/REST.php:118
Stack trace:
#0 /Users/bruce/www/personal/PicUploader/vendor/google/apiclient/src/Google/Http/REST.php(94): Google_Http_REST::decodeHttpResponse(Object(GuzzleHttp\Psr7\Response), Object(GuzzleHttp\Psr7\Request), 'Google_Service_...')
#1 /Users/bruce/www/personal/PicUploader/vendor/google/apiclient/src/Google/Task/Runner.php(176): Google_Http_REST::doExecute(Object(GuzzleHttp\Client), Object(GuzzleHttp\Psr7\Request), 'Google_Service_...')
#2 /Users/bruce/www/personal/PicUploader/vendor/google/apiclient/src/Google/Http/REST.php(58): Google_Task_Runner->run()
#3 /Users/bruce/www/personal/PicUploader/vendor/google/apiclient/src/Google/Client.php(842):  in /Users/bruce/www/personal/PicUploader/vendor/google/apiclient/src/Google/Http/REST.php on line 118
```

**Comment by jdpedrie** [comment]
Google Drive requires OAuth.

As stated [here](https://github.com/googleapis/google-api-php-client/blob/master/docs/oauth-web.md),

> OAuth 2.0 allows users to share specific data with an application while keeping their usernames, passwords, and other information private. For example, an application can use OAuth 2.0 to obtain permission from users to store files in their Google Drives.

I find the [OAuth 2.0 Playground](https://developers.google.com/oauthplayground/) to be helpful for development. You can choose the scopes you need (in this case, `https://www.googleapis.com/auth/drive`) and obtain an access token. You can easily test APIs which require OAuth without adding the entire authorization flow until you've gotten a bit farther in the development process.

```php
$client->setAccessToken($myAccessTokenFromPlayground);
```

**Comment by xiebruce** [comment]
Thank you for your reply, but what does “API Keys” use for? 

As to OAuth 2.0, I tried to use it, but got an error(when I call `$token = $client->fetchAccessTokenWithAuthCode($code);`):

Fatal error: Uncaught GuzzleHttp\Exception\ClientException: Client error: `POST https://oauth2.googleapis.com/token` resulted in a `400 Bad Request` response: { "error": "invalid_request", "error_description": "Could not determine client ID from request." } in /Users/bruce/www/personal/PicUploader/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php on line 113

I don't know what does "Could not determine client ID from request" mean. I got my clientID from here:
![image](https://user-images.githubusercontent.com/13575500/68446155-e7d12180-0216-11ea-9f8f-7469ec4edce5.png)

This is the json config file that I download:
![image](https://user-images.githubusercontent.com/13575500/68446570-0edc2300-0218-11ea-8043-fa4f32d56e36.png)

BTW, I'm in China, so all the request went through proxy, will this be the cause?
